src/query/storages/stage/src/read/row_based/batch.rs

```diff
@@ -0,0 +1,92 @@
+// Copyright 2021 Datafuse Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use databend_common_expression::BlockMetaInfo;
+
+#[derive(serde::Serialize, serde::Deserialize, Debug)]
+pub struct BytesBatch {
+    pub data: Vec<u8>,
```

Comment:
Can we use `Buffer<u8>` which could share with other local processors?

Reply:
not useful for now because we use `csv_core.read_record` to read this data only onces, data is copied by it.
 we can try it later if we do not use csv_core or when rewriting TSV.
But maybe not too much gain (only save a few copy and realloc?).

this pr forces on refacor, safty first.